src/components/SearchBar.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { useState } from "react";
+import { AutoComplete, Input } from "antd";
+import { debounce, filter, set } from "lodash";
+import {
+    SearchLookup,
+    UnpackedNormalCellLine,
+} from "../component-queries/types";
+import { GrayButton } from "./shared/Buttons";
+
+const { active } = require("../style/buttons.module.css");
+const { searchBarContainer } = require("../style/search-and-filter.module.css");
+
+interface SearchBarProps {
+    mappings: SearchLookup;
+    allCellLines: UnpackedNormalCellLine[];
+    setResults: (filteredCellLines: UnpackedNormalCellLine[]) => void;
+}
+
+const SearchBar = ({ mappings, allCellLines, setResults }: SearchBarProps) => {
+    const [options, setOptions] = useState<{ value: string }[]>([]);
+    const [currentValue, setCurrentValue] = useState<string>("");
+
+    const handleSearch = (value: string) => {
+        setCurrentValue(value);
+        if (value) {
+            const { allSearchableTerms } = mappings;
+            const nextOptions: React.SetStateAction<{ value: string }[]> = [];
+            allSearchableTerms.forEach((term: string) => {
+                if (term.toLowerCase().includes(value.toLowerCase())) {
+                    nextOptions.push({ value: term });
+                }
+            });
+            setOptions(nextOptions);
+        } else {
+            setOptions([]);
+            setResults(allCellLines);
+        }
+    };
+
+    const handleClear = () => {
+        setOptions([]);
+        setResults(allCellLines);
+        setCurrentValue("");
+    };
+
+    const onEnter = () => {
+        let selectedCellLines = new Set<number>();
+        options.forEach((option) => {
+            selectedCellLines = new Set([
+                ...selectedCellLines,
+                ...getCellLineIdsToShow(option.value),
+            ]);
```

Comment:
Can you add the elements to the initialized Set instead of creating a new Set every time?

Reply:
I could, this was a bit of a remnant from how I was doing things previously where there were a lot of duplicates that needed to be removed and this is a quick way to de-dup a list. I don't feel strongly about either method (this way verses iterating over the returned array and adding each element)